R/zchunk_LB142.ag_Fert_IO_R_C_Y_GLU.R

```diff
@@ -1,103 +1,227 @@
-    # fixing it will change the output data, causing the tests to fail,
-    # (i) open an issue on GitHub, (ii) consult with colleagues, and
-    # then (iii) code a fix:
-    #
-    # if(OLD_DATA_SYSTEM_BEHAVIOR) {
-    #   ... code that replicates old, incorrect behavior
-    # } else {
-    #   ... new code with a fix
-    # }
-    #
-    #
-    # NOTE: there are 'match' calls in this code. You probably want to use left_join_error_no_match
-    # For more information, see https://github.com/JGCRI/gcamdata/wiki/Name-That-Function
-    # NOTE: This code uses translate_to_full_table
-    # This function can be removed; see https://github.com/JGCRI/gcamdata/wiki/Name-That-Function
-    # NOTE: This code uses vecpaste
-    # This function can be removed; see https://github.com/JGCRI/gcamdata/wiki/Name-That-Function
-    # ===================================================
+    # Compile N fertilizer production by country, and adjust production so that production and consumption balance globally
+    L100.FAO_Fert_Prod_tN %>%
+      select(iso, year, prod = value) %>%
+      # Combine with fertilizer consumption, use full_join to keep all observations, such as ones only have consumption
+      full_join(select(L100.FAO_Fert_Cons_tN, iso, year, cons = value), by = c("iso", "year")) %>%
+      replace_na(list(prod = 0, cons = 0)) %>%
+      group_by(year) %>%
+      # Calculate the global total production and consumption
+      summarise(prod = sum(prod), cons = sum(cons)) %>%
+      ungroup() %>%
+      # Calculate the rate to adjust production so that global production equals consumption
+      mutate(adj = cons / prod) %>%
+      select(year, adj) ->
+      L142.ag_Fert_Prod_adj
+
+    L100.FAO_Fert_Prod_tN %>%
+      select(iso, year, value) %>%
+      left_join_error_no_match(L142.ag_Fert_Prod_adj, by = "year") %>%   # Match in the rates for adjustment
+      mutate(value = value * adj,                                        # Adjust production
+             value = value * CONV_T_MT) %>%                              # Convert unit from ton to million ton
+      select(-adj) ->
+      L142.ag_Fert_Prod_MtN_ctry_Y
+
+    # Aggregate N fertilizer productionto GCAM region level
+    L142.ag_Fert_Prod_MtN_ctry_Y %>%
+      left_join_error_no_match(iso_GCAM_regID, by = "iso") %>%           # Match in GCAM region ID
+      group_by(GCAM_region_ID, year) %>%
+      summarise(value = sum(value)) %>%                                  # Aggregate to region total
+      ungroup() %>%                                                      # Ungroup before complete
+      mutate(GCAM_commodity = aglu.FERT_NAME) %>%                             # Add GCAM commodity category for N fertilizer
+      complete(GCAM_region_ID = unique(iso_GCAM_regID$GCAM_region_ID),
+               GCAM_commodity, year, fill = list(value = 0)) ->          # Fill in missing region with 0
+      L142.ag_Fert_Prod_MtN_R_Y
+
+    # Aggregate N fertilizer consumption to GCAM region level
```

Comment:
These seem to be the exact same steps as lines 72-81. Worth making a function and calling it twice?